src/app/services/tasksApi.ts

```diff
@@ -1,14 +1,26 @@
-import task from '@/interfaces/task.type';
+import task, { updateTaskDetails } from '@/interfaces/task.type';
 import { api } from './api';
-import { MINE_TASKS_URL } from '@/constants/url';
+import { MINE_TASKS_URL, TASKS_URL } from '@/constants/url';
 
 type TasksQueryResponse = { message: string; tasks: task[] };
+type TasksCreateMutationResponse = { message: string; task: task };
+type TaskRequestPayload = { task: updateTaskDetails; id: string };
 
 export const tasksApi = api.injectEndpoints({
     endpoints: (builder) => ({
         getAllTasks: builder.query<TasksQueryResponse['tasks'], void>({
             query: () => '/tasks',
-            providesTags: ['Tasks'],
+            providesTags: (result) =>
+                result
+                    ? [
+                          ...result.map(({ id }) => ({
+                              type: 'Tasks' as const,
+                              id,
+                          })),
+                          { type: 'Tasks' },
+                      ]
+                    : ['Tasks'],
```

Comment:
NIT: Just an opinion can we use if else? or can we de-structure this in a variable then use it here

Reply:
We can use `if.. else` but this is a more concise syntax. Also used by the docs.